contrib/coco/tests/test_convert.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+#
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from pathlib import Path
+
+import numpy as np
+from pycocotools.coco import COCO
+
+from rikai.types import Mask
+
+
+def test_covert_segmentation():
+    coco = COCO(str(Path(__file__).parent / "testdata" / "coco.json"))
+
+    for image_id in coco.imgs:
+        img = coco.loadImgs(image_id)[0]
+        # print(img)
+        height, width = img["height"], img["width"]
+        ann_ids = coco.getAnnIds(imgIds=image_id)
+        for ann in coco.loadAnns(ann_ids):
+            if ann["iscrowd"] == 0:
+                mask = Mask.from_polygon(
+                    ann["segmentation"], height=height, width=width
+                )
+                # TODO: currently, it has disparity of the conversion between
```

Comment:
why is this disparity happening?
is this not important to address right now?

Reply:
So, in [coco](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/common/maskApi.c#L162),  it uses its own implementation to fill the polygon, which is different to `PIL / OpenCV`. 

I am hesitate to actually use coco's implementation.  This `check_coco.py` calculates the iou between coco algorithm and PIL for the same polygon. Most of them are fall into `[0.9-1.0]` range, but still have some in `[0.7-0.8]` range. 

But since we store the polygon directly in the data, so it is lossless, and the end user can just use `pycocotools` to render this polygon again if desired.

Reply:
got it